stats/stats.py

```diff
@@ -0,0 +1,324 @@
+            'count': {
+                'modifiers': [],
+                'granulations': {
+                    'issued_at': ['year', 'month', 'week', 'day'],
+                    'paid_at': ['year', 'month', 'week', 'day'],
+                    'currency': None
+                }
+            },
+            'amount': {
+                'modifiers': ['average', 'total'],
+                'granulations': {
+                    'issued_at': ['year', 'month', 'week', 'day'],
+                    'paid_at': ['year', 'month', 'week', 'day'],
+                    'currency': None
+                }
+            },
+            'paid_date': {
+                'modifiers': ['average'],
+                'granulations': {
+                    'issued_at': ['year', 'month', 'week', 'day'],
+                    'paid_at': ['year', 'month', 'week', 'day'],
+                    'currency': None
+                }
+            },
+            'overdue_payments': {
+                'modifiers': [],
+                'granulations': {
+                    'issued_at': ['year', 'month', 'week', 'day'],
+                    'paid_at': ['year', 'month', 'week', 'day'],
+                    'currency': None
+                }
+            }
+        },
+        'subscriptions': {
+            'estimated_income':{
+                'modifiers': ['include_unused_plans'],
+                'granulations': {
+                    'billing_date': ['year', 'month'],
+                    'plan': None
+                }
+            }
+        }
+    }
+
+    def validate(self):
+        if self.queryset.model != Transaction and self.queryset.model != Invoice \
+                and self.model != Subscription:
+            raise ValueError('invalid model : choices are: Transaction, Invoice, Subscription')
+        if self.queryset.model == Invoice:
+            model = 'documents'
+        elif self.queryset.model == Transaction:
+            model = 'transactions'
+        else:
+            model = 'subscriptions'
+        if self.result_type is None:
+            raise ValueError('Select a result type: choices are: amount, count, overdue_payments, '
+                             'estimated_income')
+
+        for key, value in self.result_types.iteritems():
+            if key != model:
+                continue
+            if self.result_type not in value:
+                raise ValueError('Invalid result type!')
+            for result_type, values in value.iteritems():
+                if result_type == self.result_type:
+                    if self.granulations['value'] not in values:
+                        raise ValueError('Invalid granulation!')
+                    elif not values['modifiers']:
+                        if self.modifier is not None:
+                            raise ValueError("This result type doesn't have modifiers!")
+                    elif self.modifier is None:
+                            raise ValueError('Select a modifier!')
```

Comment:
As a general advice, exception messages should be impersonal. Something like this would be better:
```
raise ValueError('The modifier argument is required.')
```